packages/api-aco/src/folder/onFolderBeforeDeleteFm.hook.ts

```diff
@@ -0,0 +1,43 @@
+import WebinyError from "@webiny/error";
+import { AcoContext } from "~/types";
+
+export const onFolderBeforeDeleteFmHook = (context: AcoContext) => {
+    context.aco.folder.onFolderBeforeDelete.subscribe(async ({ folder }) => {
+        try {
+            const { id, type } = folder;
+
+            /**
+             * Exit if the folder type is not related to File Manager
+             */
+            if (type !== "FmFile") {
+                return;
+            }
+
+            const [files] = await context.fileManager.listFiles({
+                where: {
+                    location: {
+                        folderId: id
+                    }
+                },
+                limit: 1
+            });
+
+            if (files.length === 0) {
+                return;
+            }
+
+            throw new WebinyError(
+                "Error: delete all child folders and files before proceeding.",
```

Comment:
Messages should be just plain messages, without `Error:` prefix. It's obvious it's an error, that's why we're throwing it.